pkg/internal/validate/valid_create_action_args_test.go

```diff
@@ -0,0 +1,216 @@
+package validate_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/fixtures"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/utils"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/validate"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/stretchr/testify/require"
+)
+
+func TestForDefaultValidCreateActionArgs(t *testing.T) {
+	// given:
+	args := fixtures.DefaultValidCreateActionArgs()
+
+	// when:
+	err := validate.ValidCreateActionArgs(&args)
+
+	// then:
+	require.NoError(t, err)
+}
+
+func TestWrongCreateActionArgs(t *testing.T) {
+	tests := map[string]struct {
+		args wdk.ValidCreateActionArgs
+	}{
+		"IsSendWith is set even though there is no 'send-with' txs in options": {
+			args: func() wdk.ValidCreateActionArgs {
+				args := fixtures.DefaultValidCreateActionArgs()
+				args.IsSendWith = true
+				args.Options.SendWith = []wdk.TXIDHexString{}
+				return args
+			}(),
```

Comment:
😕 eeee what's the point of having this in function that is called just after declaration?

Maybe a better idea would be to accept an args as and input and call this function with the default args in the test?

```suggestion
			args: func(args wdk.ValidCreateActionArgs) wdk.ValidCreateActionArgs {
				args.IsSendWith = true
				args.Options.SendWith = []wdk.TXIDHexString{}
				return args
			},
```

Reply:
They were just generator functions which produced args - they were capable of generating "any" args.
- Your suggestion modifies the default args, provided once in the test code. 
- Because - for now - all off the tests rely on the default config - I've applied your suggestion ✅ 
- But I don't think that it was serious enough to hit "change request" - just another approach.